libs/s3/s3_versioning_common_test_lib.py

```diff
@@ -20,41 +20,343 @@
+    :return: tuple, returns multiple values such as access_key, secret_key and S3 objects
+        which are required to perform further operations.
     """
-    rest_obj = S3AccountOperationsRestAPI()
-    LOG.info("Creating account with name %s and email_id %s",
-             account_name, email_id)
+    LOG.info("Creating user with name %s and email_id %s",
+             user_name, email_id)
     new_user = rest_obj.create_s3_account(user_name=user_name, 
                                           email_id=email_id,
                                           passwd=password)
     assert_utils.assert_true(new_user[0], new_user[1])
     access_key = new_user[1]["access_key"]
     secret_key = new_user[1]["secret_key"]
     del rest_obj
-    LOG.info("Successfully created the S3 account")
-    s3_obj = s3_versioning_test_lib.S3VersioningTestLib(access_key=access_key, 
-                                                        secret_key=secret_key,
-                                                        endpoint_url=S3_CFG["s3_url"])
+    LOG.info("Successfully created the S3 user")
+    s3_obj = S3VersioningTestLib(access_key=access_key, 
+                                 secret_key=secret_key,
+                                 endpoint_url=S3_CFG["s3_url"])
     response = (s3_obj, access_key, secret_key)
     return response
+
+def check_list_object_versions(s3_ver_test_obj: S3VersioningTestLib,
+                               bucket_name: str = None,
+                               list_params: dict = None,
+                               expected_versions: dict = None,
+                               expected_flags: dict = None,
+                               expected_error: str = None) -> None:
+    """
+    List all the versions and delete markers present in a bucket and verify the output
+
+    :param s3_ver_test_obj: S3VersioningTestLib object to perform S3 versioning calls
+    :param bucket_name: Bucket name for calling List Object Versions
+    :param expected_versions: dict containing list of versions, delete markers and flags
+        
+        Expected format of the dict -
+            {"<key_name1": "versions": {"version_id1": "etag1"} ...},
+                           "delete_markers": ["dm_version_id1", ...],
+                           "is_latest": "version_id1",
+            ...
+            }
+    """
+    LOG.info("Fetching bucket object versions list")
+    try:
+        if list_params:
+            list_response = s3_ver_test_obj.list_object_versions(bucket_name=bucket_name,
+                                                                 **list_params)
+        else:
+            list_response = s3_ver_test_obj.list_object_versions(bucket_name=bucket_name)
+    except CTException as error:
+        self.log.error(error)
+        list_error = error
+    
+    if expected_error is not None:
+        assert_utils.assert_in(expected_error, list_error.message, list_error)
+        return
```

Comment:
This return will refrain execution of other code in matching exception. In non matching it's a dead code.

Reply:
Added to skip evaluating ListObjectVersions response contents in scenarios where assertion succeeds.